File: _scratch/run/backspace_string_compare/sol.js
/**
 * @param {string} s
 * @param {string} t
 * @return {boolean}
 */
var backspaceCompare = function(s, t) {
    const build = (x) => {
        const st = [];
        for (const c of x) {
            if (c === '#') { if (st.length) st.pop(); }
            else st.push(c);
        }
        return st.join('');
    };
    return build(s) === build(t);
};
const _l=require('fs').readFileSync(0,'utf8').split('\n');const _T=+_l[0];const _o=[];
for(let _i=0;_i<_T;_i++){const _s=(_l[1+2*_i]||'').replace(/\r$/,'');const _t=(_l[2+2*_i]||'').replace(/\r$/,'');_o.push(backspaceCompare(_s,_t)?'true':'false');}
console.log(_o.join('\n'));

File: _scratch/run/backspace_string_compare/sol.py
class Solution:
    def backspaceCompare(self, s, t):
        def build(x):
            st = []
            for c in x:
                if c == "#":
                    if st: st.pop()
                else:
                    st.append(c)
            return "".join(st)
        return build(s) == build(t)
import sys
_l=sys.stdin.read().split('\n')
_T=int(_l[0]);_o=[]
for _i in range(_T):
    _s=_l[1+2*_i] if 1+2*_i<len(_l) else ""
    _t=_l[2+2*_i] if 2+2*_i<len(_l) else ""
    _o.append('true' if Solution().backspaceCompare(_s,_t) else 'false')
print('\n'.join(_o))